src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -167,12 +176,12 @@ func (c *IdleTimingConn) SetDeadline(t time.Time) error {
 }
 
 func (c *IdleTimingConn) SetReadDeadline(t time.Time) error {
-	c.readDeadline = &t
+	atomic.StoreInt64(&c.readDeadline, t.UnixNano())
```

Comment:
@myleshorton Switched to using atomic.StoreInt64 to avoid concurrency issues on writing/reading the deadline.